app/views/ExploreDeep/ActionCell/ProjectJoinModal/index.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { useCallback } from 'react';
+import {
+    _cs,
+    isDefined,
+} from '@togglecorp/fujs';
+import { useMutation, gql } from '@apollo/client';
+import {
+    Modal,
+    Button,
+    TextArea,
+    useAlert,
+    PendingMessage,
+} from '@the-deep/deep-ui';
+import {
+    ObjectSchema,
+    useForm,
+    removeNull,
+    requiredStringCondition,
+    getErrorObject,
+    lengthGreaterThanCondition,
+} from '@togglecorp/toggle-form';
+
+import {
+    JoinProjectMutation,
+    JoinProjectMutationVariables,
+} from '#generated/types';
+import { transformToFormError } from '#base/utils/errorTransform';
+import NonFieldError from '#components/NonFieldError';
+
+import styles from './styles.css';
+
+interface ProjectJoinFields {
+    reason: string;
+}
+
+const JOIN_PROJECT = gql`
+    mutation JoinProject(
+        $reason: String!,
+        $project: String!,
+    ) {
+        joinProject(data: {
+            reason: $reason,
+            project: $project,
+        }) {
+            ok,
+            errors,
+        }
+    }
+`;
+
+type FormType = Partial<ProjectJoinFields>;
+type FormSchema = ObjectSchema<FormType>;
+type FormSchemaFields = ReturnType<FormSchema['fields']>;
+
+const schema: FormSchema = {
+    fields: (): FormSchemaFields => ({
+        reason: [
+            requiredStringCondition,
+            lengthGreaterThanCondition(100),
+        ],
+    }),
+};
+
+const defaultFormValue: FormType = {};
+
+interface Props {
+    className?: string;
+    projectId: string;
+    onModalClose: () => void;
+    onJoinRequestSuccess: () => void;
+}
+
+function ProjectJoinModal(props: Props) {
+    const {
+        className,
+        projectId,
+        onModalClose,
+        onJoinRequestSuccess,
+    } = props;
+
+    const {
+        pristine,
+        value,
+        error: riskyError,
+        setFieldValue,
+        validate,
+        setError,
+    } = useForm(schema, defaultFormValue);
+    const alert = useAlert();
+
+    const error = getErrorObject(riskyError);
+
+    const [
+        joinProject,
+        { loading },
+    ] = useMutation<JoinProjectMutation, JoinProjectMutationVariables>(
+        JOIN_PROJECT,
+        {
+            onCompleted: (response) => {
```

Comment:
We need to handle onError condition as well

Reply:
There should be a generic onFailure imo